ftplugin/markdown/quicklink.vim

```diff
@@ -95,4 +107,40 @@ function! OpenLinkOnCurrentLine()
   call system('open ' . expand('<cWORD>'))
 endfunction
 
+function! s:OnMarkdownLink()
+  let current_syntax_id = synIDattr(synID(line("."), col("."), 1), "name")
+  return count(s:MARKDOWN_LINK_SYNTAX_IDS, current_syntax_id) != 0
+endfunction
+
+function! s:CaptureLinkText()
+  let @a = ""
+  normal "ayi]
+  return escape(getreg('a'), '&')
+endfunction
+
+function! s:OpenWithNetrw() 
+  call netrw#BrowseX(expand("<cfile>"),0) 
+endfunction
+
+function! s:OpenMarkdownLink()
+  let initial_pos = getpos('.')
+  let escaped_link_name = s:CaptureLinkText()
+  let link_target_pattern = '\v^\['.escaped_link_name.'\]: (%(ftp[s]?|http[s]?):\/\/\S+)>'
+  if search(link_target_pattern, 'e')
+    call s:OpenWithNetrw()
+  endif
+  call setpos('.', initial_pos)
+endfunction
+
+function! s:MarkdownAwareGX()
+  if s:OnMarkdownLink()
+    call s:OpenMarkdownLink()
+  else
+    call netrw#BrowseX(expand("<cfile>"),0)
```

Comment:
That sounds good. Looks like you've done that in the `s:OpenWithNetrw` function above, but you don't seem to be calling that function here. Am I missing something?